### DISCUSSION TITLE
Paginator not working on manual list widget

### DISCUSSION THREAD

**Discussion by Tony-S** [wintercms/winter#924] [Q&A]
I have manually created a list widget to list orders for a user in the backend. It is manually created as it actually fetches orders for anyone in the user's company. All is working except the paginator. It paginates the results but the next and previous page links give an error:
"A widget with class name 'list' has not been bound to the controller" on line 606 of <backend path>/modules/backend/classes/Controller.php"
The data request on the paginator links is data-request="list::onPaginate"
The code in the partial is below. The commented out bits work but are not relevant to the issue. Can you advise how to solve this please?
Thanks.
```php
<?php

use Acme\Order\Models\Order;

//$companyUsers = $model->getAllUserIdsForCompany();
$config = $this->makeconfig('$/acme/order/models/order/columns_for_user.yaml');
$config->model = new Order;
$widget = $this->makeWidget('Backend\Widgets\Lists', $config);
$widget->recordUrl = 'acme/order/orders/view/:id';
$widget->showPagination = true;
$widget->recordsPerPage = 3;
$widget->noRecordsMessage = 'No orders';
//$widget->bindEvent('list.extendQuery', function ($query) use ($companyUsers) {
//    $query->whereIn('user_id', $companyUsers);
//});
$widget->bindToController();
echo $widget->render();
```

**Comment by LukeTowers** [comment]
PHP is stateless. If you create your list widget manually in a given context (i.e. to display it) but don't ensure that it is created and bound to the controller for the AJAX request that is being sent for the `list::onPaginate` then you'll run into that issue. Please share your entire controller, not just the snippet where you create the widget.
